pulp_container/app/registry_api.py

```diff
@@ -988,23 +988,31 @@ def put(self, request, path, pk=None):
                 objs=thru, ignore_conflicts=True, batch_size=1000
             )
 
-        tag = models.Tag(name=pk, tagged_manifest=manifest)
-        try:
-            tag.save()
-        except IntegrityError:
-            tag = models.Tag.objects.get(name=tag.name, tagged_manifest=manifest)
-            tag.touch()
+        # a manifest cannot tagged by its digest - an identifier specified in the 'pk' parameter
+        if not pk.startswith("sha256:"):
+            tag = models.Tag(name=pk, tagged_manifest=manifest)
+            try:
+                tag.save()
+            except IntegrityError:
+                tag = models.Tag.objects.get(name=tag.name, tagged_manifest=manifest)
+                tag.touch()
+
+            tags_to_remove = models.Tag.objects.filter(
+                pk__in=repository.latest_version().content.all(), name=tag
+            ).exclude(tagged_manifest=manifest)
+            add_content_units = [str(tag.pk), str(manifest.pk)]
```

Comment:
how about you define add and remove content units empty list before line 992 and then you just append to it when needed

Reply:
this way you will avoid the else block